api.py

```diff
@@ -0,0 +1,277 @@
+            }
+        ]
+        result = self._get_meta_response({"routes": routes})
+        return jsonify(result), 200
+
+    def list_spiders(self):
+        spiders = self.process.spider_loader.list()
+        return jsonify({"spiders": spiders, "status": "success"}), 200
+
+    def run_spider(self):
+        post_data = request.get_json()
+        spider = post_data.get('spider')
+        if spider == 'who_iris':
+            spider = who_iris_spider.WhoIrisSpider()
+        elif spider == 'nice':
+            spider = nice_spider.NiceSpider()
+        else:
+            return '', 400
+        spider_id = str(uuid.uuid4())
+        self.process.crawl(spider, uuid=spider_id)
+        crawl = self.process.join()
+        self.database.insert_spider(spider.name, spider_id)
+        crawl.addBoth(self.on_success)
+        return jsonify({
+            "data": {
+              "status": "running",
+              "spider": spider.name,
+              "_id": spider_id
+            }
+        }), 200
+
+    def on_success(self, data):
+        self.database._close_all_spiders()
+
+    def close_spider(self, spider_id):
+        for crawl in self.process.crawlers:
+            if crawl.spider.uuid == uuid:
+                crawl.stop()
+                return jsonify({
+                    "data":
+                        {"status": "success", "_id": spider_id}
+                    }), 200
+        return '', 400
+
+    def list_crawls(self):
+        crawls = self.process.crawlers
+        running_spiders = []
+        for crawl in crawls:
+            start_time = crawl.stats.get_value('start_time')
+            spider = {
+                '_id':
+                    crawl.spider.uuid,
+                'spider':
+                    crawl.spider.name,
+                'strart_time':
+                    start_time,
+                'total_time':
+                    str(datetime.now() - start_time),
+                'item_dropped':
+                    crawl.stats.get_value('item_dropped_count'),
+                'item_scraped':
+                    crawl.stats.get_value('item_scraped_count'),
+                'total_requests':
+                    crawl.stats.get_value('downloader/request_count'),
+            }
+
+            running_spiders.append(spider)
+        finished_spiders = []
+        for spider in self.database.get_finished_crawls():
+            finished_spiders.append(spider)
+        spiders = {"crawling": running_spiders, "finished": finished_spiders}
+        return jsonify({"data": {"spiders": spiders}}), 200
+
+    def stop(self):
+        self.process.stop()
+        return jsonify({"data": {"status": "success"}}), 200
+
+    def export_db(self):
+        articles_rows = self.database.get_articles()
+        articles = []
+        now = datetime.now()
+        for row in articles_rows:
+            articles.append({
+                'title': row[0],
+                'file_hash': row[1],
+                'url': row[2],
```

Comment:
any way where row can contain less than 3 elements? 

one suggestion to make this slightly more readable is to write `for title, file_hash, url in article_rows`